card_generator/api/v1/cards/views.py

```diff
@@ -44,3 +47,33 @@ def fields(self, request, **kwargs):
         """List all variable fields present in the template where the user can provide a value."""
         data = {"fields": self.get_object().get_fields()}
         return Response(data=data)
+
+    @action(methods=["post"], detail=True, url_path="merge-cards")
+    def merge_cards(self, request, **kwargs):
+        batch_id = request.data.get("batch_id")
+        if not batch_id:
+            return Response(status=400, data={"message": "Missing 'batch_id'."})
+        if not isinstance(batch_id, str) and not isinstance(batch_id, int):
+            return Response(status=400, data={"message": "Invalid 'batch_id'."})
+
+        if isinstance(batch_id, str) and not batch_id.isalnum():
+            return Response(status=400, data={"message": "Invalid 'batch_id'."})
+
+        client = QueueCardsClient(
+            server_root=settings.PDS_SPP_SERVER_ROOT,
+            username=settings.PDS_SPP_USERNAME,
+            password=settings.PDS_SPP_API_TOKEN,
+            db_name=settings.PDS_SPP_DB_NAME,
+        )
+        record = client.get_id_queue_pdfs(batch_id=batch_id)
```

Comment:
Maybe this should be delayed as well, as it could take time. Maybe we could just check that the queue exists and then do the fetching in a task.

Reply:
Makes sense. will update the code. Thanks @jeremi